src/main/java/com/ibm/disni/rdma/verbs/impl/NatPollCqCall.java

```diff
@@ -74,11 +74,7 @@ public void set(IbvCQ cq, IbvWC[] wcList, int ne) {
 		this.ne = ne;
 		
 		this.csize = wcList.length*IbvWC.CSIZE;
-		if (cmd != null){
-			cmd.free();
-			cmd = null;
-		}		
-		this.cmd = memAlloc.allocate(csize, MemoryAllocation.MemType.DIRECT, this.getClass().getCanonicalName());
+		this.cmd = memAlloc.cleanOrAllocate(cmd, csize);
```

Comment:
I think we should not do this. I know we currently cache SVC objects in RdmaVerbsImpl and following from there it's possible that new requests are re-using an existing object but with the large or smaller WR list. But I think both is a design mistake actually. We shouldn't cache the SVC objects (https://github.com/zrlio/disni/blob/master/src/main/java/com/ibm/disni/rdma/verbs/impl/RdmaVerbsNat.java#L58) because those objects are meant to be cached by the app for re-occuring operations of the exact same type. If do so we can eliminate the ability to re-set the WRs to a completely new list. We can also eliminate the check whether the size has increased, we don't need to allocate methods in MemoryAllocater, etc. We can throw out all that code path.

Reply:
This basically to reuse existing buffer if there's one of such size in cache. It shouldn't affect if there would be new SCV object or cached one.

Reply:
Here is what I suggest: we remove the cleanOrAllocate for this PR. Then I guess the PR can be accepted and merged. After that let me create another PR that removes the SVC object caching and cleans up a bunch of other stuff related to that part. This will substantially simplify the codebase. We are always seeking for the simplest solution that satisfies the required needs.

Reply:
Done

Reply:
@patrickstuedi ping